src/envs/blocks.py

```diff
@@ -433,10 +453,12 @@ def _Stack_policy(self, state: State, memory: Dict,
     def _PutOnTable_policy(self, state: State, memory: Dict,
                            objects: Sequence[Object], params: Array) -> Action:
         del state, memory, objects  # unused
-        # Un-normalize parameters to actual table coordinates
+        # De-normalize parameters to actual table coordinates.
         x_norm, y_norm = params
-        x = self.x_lb + (self.x_ub - self.x_lb) * x_norm
-        y = self.y_lb + (self.y_ub - self.y_lb) * y_norm
+        x = self.x_lb + self.block_size / 2 + (self.x_ub - self.x_lb -
```

Comment:
why does the second part of this expression not need to use the half_size as well? we're trying to normalize to within `[self.x_lb + half_size, self.x_ub - half_size]` right?

Reply:
oh, so `(self.x_ub - half_size) - (self.x_lb + half_size) = self.x_ub - self.x_lb - self.block_size` i see. this is really confusing; would it be better to change x_lb/x_ub/y_lb/y_ub to factor in the half sizes?